.github/workflows/OCV-iOS-Framework.yaml

```diff
@@ -0,0 +1,53 @@
+name: OCV iOS Framework
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - '.github/workflows/OCV-iOS-Framework.yaml'
+  workflow_call:
+  schedule:
+  - cron: "0 0 * * *"
+
+env:
+  PR_AUTHOR: ${{ github.event.pull_request.user.login }}
+  PR_AUTHOR_FORK: ${{ github.event.pull_request.head.repo.full_name }}
+  SOURCE_BRANCH_NAME: ${{ github.head_ref }}
+  TARGET_BRANCH_NAME: ${{ github.base_ref }}
+
+jobs:
+  Build:
+    runs-on: macOS-M1
+    steps:
+    - name: Setup infra environment
+      if: ${{ github.event.repository.name == 'ci-gha-workflow' }}
+      run: echo "TARGET_BRANCH_NAME=4.x" >> $GITHUB_ENV
+    - name: PR info
+      run: |
+        echo "PR Author: ${{ env.PR_AUTHOR }}"
+        echo "PR Author fork: ${{ env.PR_AUTHOR_FORK }}"
+        echo "Source branch name: ${{ env.SOURCE_BRANCH_NAME }}"
+        echo "Target branch name: ${{ env.TARGET_BRANCH_NAME }}"
+    - name: Clean
+      run: cd ${{ github.workspace }} && find . -mindepth 1 -delete
+    - name: Fetch opencv
+      run: cd ${{ github.workspace }} && git clone --branch ${{ env.TARGET_BRANCH_NAME }} --reference $GIT_CACHE/opencv.git https://github.com/opencv/opencv.git opencv
+    - name: Merge opencv with ${{ env.SOURCE_BRANCH_NAME }} branch
+      if: ${{ github.event.repository.name != 'ci-gha-workflow' }}
+      run: |
+        cd ${{ github.workspace }}/opencv
+        git config user.email "opencv.ci"
+        git config user.name "opencv.ci"
+        git pull -v "https://github.com/${{ env.PR_AUTHOR_FORK }}" "${{ env.SOURCE_BRANCH_NAME }}"
+    - name: Build OpenCV iOS Framework
+      run: |
+        cd ${{ github.workspace }} && python3 opencv/platforms/ios/build_framework.py ios
+    - name: Create Artifact
+      run: cd ${{ github.workspace }}/ios && zip -r -9 -y OpenCViOSFramework.zip opencv2.framework
+    - name: Upload Artifact
+      uses: actions/upload-artifact@v3
+      if: github.event_name == 'schedule'
```

Comment:
There is "debug" option for workflows. Could you add upload on debug run too?

Reply:
Added `if` condition for this step to keep an archive when a debug logging is turned on.